### DISCUSSION TITLE
Validation, List of Value, Multiple Selection: How?

### DISCUSSION THREAD

**Discussion by maquejp** [PHPOffice/PhpSpreadsheet#2041] [Q&A] [answered]
Hello,
Is it possible to add validation as dropdown with multi selection?
I have this for the moment, but the user want to be able to select multiple values.
`            $validation = $sheet->getCell($column . $c)->getDataValidation();
            $validation->setType(DataValidation::TYPE_LIST)
                ->setErrorStyle(DataValidation::STYLE_INFORMATION)
                ->setAllowBlank(false)
                ->setShowInputMessage(true)
                ->setShowErrorMessage(true)
                ->setShowDropDown(true)
                ->setErrorTitle('Input Error')
                ->setError('Value is not in the list')
                ->setPromptTitle('Please pick in the list')
                ->setPrompt('Please pick a value in the list')
                ->setFormula1('"'.$lovPreps.'"');`

I have seen workaround in XLS using VBA but as far as I know I cannot embed macros with PSSheet

**Comment by MarkBaker** [comment, marked as answer]
No, I'm afraid that lists are for selection of a single entry only. This isn't possible in Excel without the use of VBA; and PHPSpreadsheet does not support macros

**Reply by callmehuohuo** [reply]
If this value is large, the generated excel will display blank and cannot be selected. What is the reason

**Reply by MarkBaker** [reply]
MS Excel has a limit to the size of a list; but you can bypass that limit by setting the list as formula that references a range of cell values. This is documented in [Setting data validation on a cell](https://phpspreadsheet.readthedocs.io/en/latest/topics/recipes/#setting-data-validation-on-a-cell)
